endpoint/spec.md

```diff
@@ -657,6 +663,33 @@ Example:
 }
 ```
 
+When this specification, and the [message specification](../message/spec.md),
+are used with specifications such as [CloudEvents](https://cloudevents.io),
+where a semantically unique identifier is used in a runtime message (e.g.
+CloudEvent's `type` attribute), it is STRONGLY RECOMMENDED that the
+`messageid` values of the message definitions for an Endpoint match that
+unique identifier and therefore be unique across all messages within the
+`messages` collection and the messages referenced by the `messagegroups`
+attribute. This will allow for an easily "lookup" from an incoming runtime
+message to it's related message definition.
+
+However, there are times when this is not possible. For example, take the case
+where an Endpoint might have the same semantic message defined twice, once for
+a JSON serialization and once for an XML serialization. Using the same
+`messageid` value is not possible (even though the CloudEvent `type` attribute
+would be the same for both runtime messages), so one (or both) message
+definition's `messageid` values might not match the runtime message's `type`
+value. In those cases, finding the appropriate message definition will need to
+be done via examination of some other metadata - such as the message's
+`envelopemetadata.type` value along with its `envelopeoptions.format` value.
+These details are of scope for this specification to define and are left as
+an implementation detail.
+
+Implementations MAY choose to generate an error if it detects duplicate
+`messageid` values across the `messages` collection message definitions and
+the `messagegroups` referenced message definitions, if that is the desired
+constraints for their users.
+
```

Comment:
@jskeet this was the "may generate an error" text you asked for a few weeks ago.